widgets/linechart/formatter.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linechart
+
+import (
+	"fmt"
+	"math"
+	"time"
+)
+
+// durationSingleUnitPrettyFormat returns the pretty format in one single
+// unit for a time.Duration, the different returned unit formats
+// are: nanoseconds, microseconds, milliseconds, seconds, minutes
+// hours, days.
+func durationSingleUnitPrettyFormat(d time.Duration, decimals int) string {
+	// Check if the duration is less than 0.
+	prefix := ""
+	if d < 0 {
+		prefix = "-"
+		d = time.Duration(math.Abs(d.Seconds()) * float64(time.Second))
```

Comment:
It works ok (https://play.golang.org/p/o2nfSMCMS3A). I'll add a test!

Reply:
Ah of course, it works because duration.Seconds() returns a float that can represent sub-second values. I didn't realize that. Thank you for the proof and the unit test.